rotkehlchen/chain/evm/decoding/rainbow/decoder.py

```diff
@@ -0,0 +1,261 @@
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=fee_asset,
+            amount=fee_amount,
+            location_label=sender,
+            notes=f'Spend {fee_amount} {fee_asset.resolve_to_asset_with_symbol().symbol} as Rainbow Swap fees',  # noqa: E501
+            counterparty=CPT_RAINBOW_SWAPS,
+            address=self.router_address,
+        )
+
+        # Add action item to transform the transfer event
+        # of the purchased token to the user's wallet
+        amount = asset_normalized_value(in_amount, self._resolve_asset(in_asset))
+        context.action_items.append(ActionItem(
+                action='transform',
+                from_event_type=HistoryEventType.RECEIVE,
+                from_event_subtype=HistoryEventSubType.NONE,
+                asset=in_asset,
+                amount=amount,
+                to_event_type=HistoryEventType.TRADE,
+                to_event_subtype=HistoryEventSubType.RECEIVE,
+                to_counterparty=CPT_RAINBOW_SWAPS,
+                to_notes=f'Receive {amount} {in_asset.resolve_to_asset_with_symbol().symbol} as the result of a Rainbow Swap',  # noqa: E501
+                address=self.router_address,
+            ))
+
+        return DecodingOutput(event=fee_event)
+
+    def _decode_swap_native_to_token(self, context: DecoderContext) -> DecodingOutput:
+        """
+        Decodes a swap transaction from native token to another token.
+        This function the fee charged using the wrapperd nativa token deposit log
+        and the token being bought in a swap transaction where the native token is swapped for another token.
+        """  # noqa: E501
+
+        buy_asset = None
+        fee_raw = None
+        buy_amount = None
+        for log in context.all_logs:
+            # Find the fee charged using the deposit log on wrapped native token contract
+            if log.topics[0] == WETH_DEPOSIT_TOPIC:
+                fee_raw = context.transaction.value - int.from_bytes(log.data)
+            # Find the buy token using the transfer log
+            if log.topics[0] == ERC20_OR_ERC721_TRANSFER and bytes_to_address(log.topics[2]) == context.transaction.from_address:  # noqa: E501
+                buy_asset = self.base.get_or_create_evm_token(log.address)
+                buy_amount = int.from_bytes(log.data)
+
+            if buy_asset and buy_amount and fee_raw:
+                break
+
+        if not (buy_asset and fee_raw and buy_amount):
+            return DEFAULT_DECODING_OUTPUT
+
+        return self._decode_swap(
+            context=context,
+            fee_raw=fee_raw,
+            in_asset=buy_asset,
+            in_amount=buy_amount,
+        )
+
+    def _decode_swap_token_to_native(self, context: DecoderContext) -> DecodingOutput:
+        tx_log = context.tx_log
```

Comment:
this needs a docstring explaining what it does, also don't assing a variable only for tx_log